onnxruntime/core/optimizer/conv_activation_fusion.cc

```diff
@@ -49,11 +49,17 @@ bool ConvFusionDataTypeCheck(const Node& conv_node) {
   // Assess the support level for the other compatible EPs and if they also
   // only support float, remove the EP check altogether.
   const std::string_view node_ep = conv_node.GetExecutionProviderType();
-  if (node_ep == kCudaExecutionProvider || node_ep == kCpuExecutionProvider) {
+  if (node_ep == kCudaExecutionProvider) {
     if (!HasElementDataType(*conv_node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT)) {
       return false;
     }
   }
+  if (node_ep == kCpuExecutionProvider) {
+    if (!HasElementDataType(*conv_node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT) &&
+        !HasElementDataType(*conv_node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT16)) {
```

Comment:
>TensorProto_DataType_FLOAT16 [](http://example.com/codeflow?start=71&length=28)

CPU EP does NOT support fp16 all the time. And the supports differs on different hardware platforms. It may also change in later revisions. Currently we have static compilation flag and also dynamic feature check functions defined for this support. Use them!